compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -721,13 +721,11 @@ impl<'a, 'tcx> CastCheck<'tcx> {
         use rustc_middle::ty::cast::IntTy::*;
 
         if self.cast_ty.is_dyn_star() {
```

Comment:
Honestly, this should probably be `self.expr_ty.is_dyn_star() || self.cast_ty.is_dyn_star()` to fix the diagnostic regression from 46ecb23198fa in `tests/ui/dyn-star/dyn-to-rigid.rs`.